clang/lib/Frontend/TextDiagnostic.cpp

```diff
@@ -1251,7 +1251,7 @@ highlightLines(StringRef FileData, unsigned StartLineNumber,
     unsigned LineLength = 0;
     for (unsigned I = 0; I <= Spelling.size(); ++I) {
       // This line is done.
-      if (isVerticalWhitespace(Spelling[I]) || I == Spelling.size()) {
+      if (I == Spelling.size() || isVerticalWhitespace(Spelling.at(I))) {
```

Comment:
The bug was the order of the instruction,  the `at` is not doing anything useful here here. checking the size first should be necessary and sufficient.

The interesting question is why that wasn't caught by tools?
We have a repro? ( I don't insist on having one to land that though)

Reply:
Well it would've been caught earlier if it had been using `at()` in the first place.

> We have a repro? ( I don't insist on having one to land that though)

Even though I have assertions and sanitizers (but not msan) enabled locally, the `Spelling[I]` did not crash for me or cause anything else in the `I == Spelling.size()` case.

Reply:
I won't insist but I'd rather remove the at.
It occurs to me that between the lack of warnings, and the fact that unit tests probably have color disabled, this won't be easy to test.... oh well